certora/specs/ghost.spec

```diff
@@ -0,0 +1,122 @@
+/***
+
+Spec for the Certora Verification Language. 
+See https://docs.certora.com for a complete guide.
+
+***/
+
+
+/*
+    Declaration of methods that are used in the rules. `envfree` indicates that
+    the method is not dependent on the environment (`msg.value`, `msg.sender`, `block.timestamp`).
+    Methods that are not declared here are assumed to be dependent on the
+    environment.
+*/
+methods {
+	// function in the main contract under verification 
+	function boo() external  returns (string memory) envfree;
+}
+
+/** 
+	@title Find and show a path for each method.
+	@dev This is a parametric rule - it checks all public/external methods of the contracts listed.
+**/
+rule reachability(method f)
+{
+	env e;
+	calldataarg args;
+	f(e,args); //takes into account only non reveting cases 
+	satisfy true;
+}
+
+/** 
+  @title  Define and check functions that should never revert
+	@dev  use f.selector to state which functions should not revert,e.g.f.selector == sig:balanceOf(address).selector 
+**/  
+definition nonReveritngFunction(method f) returns bool = true; 
+
+rule noRevert(method f) filtered {f -> nonReveritngFunction(f) }
+{
+	env e;
+	calldataarg arg;
+  //consider auto filtering for non-payable functions 
```

Comment:
We can change the requirement to
`require !f.selector.isPayable => e.msg.value == 0;`

Reply:
As an aside, `isPayable` does not appear in our documentation as official syntax https://docs.certora.com/en/latest/docs/cvl/expr.html
Is that documentation page maintained? I ran a Prover check and `isPayable` is a part of the syntax: https://vaas-stg.certora.com/output/85695/c6429aa2ff7943f19349ae78eda531b4?anonymousKey=57f879ad125b4c37ea0f05e477044d57c4a402f1

Reply:
done